python/fusion_engine_client/messages/configuration.py

```diff
@@ -864,8 +860,16 @@ def pack(self, buffer: bytes = None, offset: int = 0, return_buffer: bool = True
 
     def unpack(self, buffer: bytes, offset: int = 0) -> int:
         parsed = self.ConfigResponseMessageConstruct.parse(buffer[offset:])
+
         self.__dict__.update(parsed)
-        self.config_object = _conf_gen.CONFIG_MAP[parsed.config_type].parse(parsed.config_data)
+        del self.__dict__['config_length_bytes']
+        del self.__dict__['config_data']
+
+        if parsed.config_length_bytes > 0:
+            self.config_object = _conf_gen.CONFIG_MAP[parsed.config_type].parse(parsed.config_data)
+        else:
```

Comment:
So for this feature where you're supporting default parsing for invalid configuration objects, is the intent that we might send a bad value as part of a `ConfigResponseMessage` and we want to not throw a parsing error? Generally, this would only happen if there's a bug in our firmware or a discrepancy between the C++ and Python interpretation of a ConfigType. I guess I would probably prefer a parsing error to silently filling in default values.

Reply:
I was running into this specifically for `UNAVAILABLE` config responses where we intentionally omit the payload (since we're not handling the request). In that case, we _could_ include the appropriate payload for the requested config type, just with default values, but I think that is actually less clear than just not having a payload since someone might think it's actually populated.

Reply:
Sorry hit enter before I replied to that last sentence.

I tend to agree that having it populate a default struct here in Python is probably equally confusing. I only did this because the Python class doesn't actually have a `config_type` member variable, and just implicitly gets it from `config_object`. So if you tried to print out a `repr()` for an `UNAVAILABLE` message, it would crash.

Maybe a better solution is to add a `__getattr__` handler for `config_type`, and then store the deserialized type internally. I'll think about that a bit.